cmake/Version.cmake

```
@@ -14,6 +16,81 @@ ly_file_read("${LY_ROOT_FOLDER}/engine.json" tmp_json_data)
 set_property(GLOBAL PROPERTY O3DE_ENGINE_JSON_DATA ${tmp_json_data})
 unset(tmp_json_data)
 
+#! o3de_get_name_and_version_specifier: Parse the dependency name, and optional version 
+# operator and version from the supplied input string. 
+# \arg:input - the input string e.g. o3de==1.0.0
+# \arg:dependency_name - the name to the left of any optional version specifier
+# \arg:operator - the version specifier operator e.g. == 
+# \arg:version - the version specifier version e.g. 1.0.0
+function(o3de_get_name_and_version_specifier input dependency_name operator version)
+    if("${input}" MATCHES "^(.*)(~=|==|!=|<=|>=|<|>|===)(.*)$")
+        if(${CMAKE_MATCH_COUNT} GREATER_EQUAL 1)
+            set(${dependency_name} ${CMAKE_MATCH_1} PARENT_SCOPE)
+        endif()
+        if(${CMAKE_MATCH_COUNT} GREATER_EQUAL 2)
+            set(${operator} ${CMAKE_MATCH_2} PARENT_SCOPE)
+        endif()
+        if(${CMAKE_MATCH_COUNT} GREATER_EQUAL 3)
+            set(${version} ${CMAKE_MATCH_3} PARENT_SCOPE)
+        endif()
+    else()
+        # format unknown, assume it's just the dependency name
+        set(${dependency_name} ${input} PARENT_SCOPE)
+    endif()
+endfunction()
+
+#! o3de_get_version_compatible: Check if the input version is compatible based on
+# the operator and specifier version provided
+# \arg:version - input version to check  e.g. 1.0.0
+# \arg:op - the version specifier operator e.g. ==
+# \arg:specifier_version - the version part of the version specifier  e.g. 1.2.0
+# \arg:is_compatible - TRUE if version is compatible otherwise FALSE 
+function(o3de_get_version_compatible version op specifier_version is_compatible)
+    set(contains_version FALSE)
+
+    if(op STREQUAL "==" AND version VERSION_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "!=" AND NOT version VERSION_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "<=" AND version VERSION_LESS_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL ">=" AND version VERSION_GREATER_EQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "<" AND version VERSION_LESS specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL ">" AND version VERSION_GREATER specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "===" AND version STREQUAL specifier_version)
+        set(contains_version TRUE)
+    elseif(op STREQUAL "~=")
+        # compatible versions have an equivalent combination of >= and == 
+        # e.g. ~=2.2 is equivalent to >=2.2,==2.*
+        if(version VERSION_GREATER_EQUAL specifier_version)
+            string(REPLACE "." ";" specifer_version_part_list ${specifier_version})
+            list(LENGTH specifer_version_part_list list_length)
```

typo: specifer -> specifier
```suggestion
            list(LENGTH specifier_version_part_list list_length)
```